parsl/executors/workqueue/executor.py

```diff
@@ -0,0 +1,438 @@
+# from parsl.data_provider.files import File
+from parsl.executors.workqueue import workqueue_worker
+
+from work_queue import *
+
+logger = logging.getLogger(__name__)
+
+
+def WorkQueueThread(tasks={},
+                    task_queue=deque,
+                    queue_lock=threading.Lock(),
+                    tasks_lock=threading.Lock(),
+                    see_worker_output=False,
+                    port=50000,
+                    launch_cmd=None,
+                    data_dir=".",
+                    log_dir=None,
+                    full=False,
+                    password=None,
+                    password_file=None,
+                    project_name=None,
+                    env=None):
+
+    logger.debug("Starting WorkQueue Master Thread")
+
+    wq_to_parsl = {}
+    if log_dir is not None:
+        wq_debug_log = os.path.join(log_dir, "debug")
+        cctools_debug_flags_set("all")
+        cctools_debug_config_file(wq_debug_log)
+
+    logger.debug("Creating Workqueue Object")
+    try:
+        q = WorkQueue(port)
+    except Exception as e:
+        logger.error("Unable to create Workqueue object: {}", format(e))
+        raise e
+
+    if project_name:
+        q.specify_name(project_name)
+
+    if password:
+        q.specify_password(password)
+    elif password_file:
+        q.specify_password_file(password_file)
+
+    # Only write Logs when the log_dir is specified, which is most likely always will be
+    if log_dir is not None:
+        wq_master_log = os.path.join(log_dir, "master_log")
+        wq_trans_log = os.path.join(log_dir, "transaction_log")
+        if not full:
+            wq_resource_log = os.path.join(log_dir, "resource_logs")
+            q.enable_monitoring_full(dirname=wq_resource_log)
+
+        q.specify_log(wq_master_log)
+        q.specify_transactions_log(wq_trans_log)
+
+    continue_running = True
+    while(continue_running):
+        # Monitor the Task Queue
+
+        place_holder_queue = []
+
+        queue_lock.acquire()
+        queue_len = len(task_queue)
+        while queue_len > 0:
+            place_holder_queue.append(task_queue.pop())
+            queue_len -= 1
+        queue_lock.release()
+
+        # Submit Tasks
+        for item in place_holder_queue:
+            parsl_id = item["task_id"]
+            if parsl_id < 0:
+                continue_running = False
+                break
+            function_data_loc = item["data_loc"]
+            function_result_loc = item["result_loc"]
+            function_result_loc_remote = function_result_loc.split("/")[-1]
+            function_data_loc_remote = function_data_loc.split("/")[-1]
+
+            input_files = item["input_files"]
+            output_files = item["output_files"]
+
+            # TODO Make this general
+            # full_script_name = "/afs/crc.nd.edu/user/a/alitteke/parsl/parsl/executors/workqueue/workqueue_worker.py"
+            full_script_name = workqueue_worker.__file__
```

Comment:
In the current implementation of finding the script in the `bin` then using parsl to load the runner on the other side was another dependency that we were looking to remove.  These two lines of code allow us to find the file in the package, and then copy that file to the worker.